File: client/src/app/entities/product.ts
import {DataPage} from '../shared/data-page';
import {User} from './user';
import {Productstatus} from './productstatus';
import {Producttype} from './producttype';

export class Product{
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  name: string;
  photo: string;
  price: number;
  producttype: Producttype;
  productstatus: Productstatus;

  constructor(id: number= null) {
    this.id = id;
  }
}

export class ProductDataPage extends DataPage{
  content: Product[];
}

File: server/src/main/java/bit/project/server/entity/Product.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    private String name;

    private String photo;

    private BigDecimal price;

    @ManyToOne
    private Producttype producttype;

    @ManyToOne
    private Productstatus productstatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Product(Integer id) {
        this.id = id;
    }

    public Product(Integer id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;
    }
}
